django/contrib/postgres/operations.py

```diff
@@ -33,11 +40,19 @@ def database_forwards(self, app_label, schema_editor, from_state, to_state):
     def database_backwards(self, app_label, schema_editor, from_state, to_state):
         if not router.allow_migrate(schema_editor.connection.alias, app_label):
             return
-        schema_editor.execute("DROP EXTENSION %s" % schema_editor.quote_name(self.name))
+
+        if self.extension_present(schema_editor, self.name):
+            schema_editor.execute("DROP EXTENSION %s" % schema_editor.quote_name(self.name))
+
         # Clear cached, stale oids.
         get_hstore_oids.cache_clear()
         get_citext_oids.cache_clear()
 
+    def extension_present(self, schema_editor, extension):
+        cur = schema_editor.connection.cursor()
+        cur.execute("SELECT * FROM pg_extension WHERE extname = %s", [extension])
+        return cur.fetchone()
```

Comment:
Can we make this function return `bool(cur.fetchone())` rather than the entire row? 

And is it worth adding a dedicated test for this method?

Reply:
added `bool()`.

I think a test would be useful but not in its current form. The current one just tests that the query was indeed executed by the engine.  It would not catch a syntax error in the query for example. instead it might:

1. assert extension not installed using something like `extension_present()`
2. install extension
3. assert extension is installed  using something like `extension_present()`